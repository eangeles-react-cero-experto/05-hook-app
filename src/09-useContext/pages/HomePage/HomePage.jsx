import { useContext } from "react";
import { UserContext } from "../../context/UserContext.jsx";

export const HomePage = () => {
  const { user } = useContext(UserContext);

  return (
    <>
      <h1>
        HomePage <small>{user?.username}</small>
      </h1>
      <hr />

      <pre data-testid="code-block">{JSON.stringify(user, null, 4)}</pre>
    </>
  );
};
