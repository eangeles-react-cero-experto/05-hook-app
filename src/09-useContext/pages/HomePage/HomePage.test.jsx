import { render, screen } from "@testing-library/react";
import { HomePage } from "./HomePage.jsx";
import { UserContext } from "../../context/UserContext.jsx";

describe("Test HomePage", () => {
  const user = {
    username: "John Due",
    email: "john@email.com",
  };

  it("should match with snapshot", () => {
    const { container } = render(
      <UserContext.Provider value={{ user }}>
        <HomePage />
      </UserContext.Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render the component without the user", () => {
    render(
      <UserContext.Provider value={{ user: null }}>
        <HomePage />
      </UserContext.Provider>,
    );

    const preElement = screen.getByTestId("code-block");
    expect(preElement.innerHTML).toEqual("null");
  });

  it("should render the component with the user", () => {
    render(
      <UserContext.Provider value={{ user }}>
        <HomePage />
      </UserContext.Provider>,
    );

    const preElement = screen.getByTestId("code-block");
    expect(screen.findByText(user.username)).toBeTruthy();
    expect(screen.findByText(user.email)).toBeTruthy();
  });
});
