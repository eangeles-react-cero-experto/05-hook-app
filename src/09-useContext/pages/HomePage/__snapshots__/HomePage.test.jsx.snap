// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomePage should match with snapshot 1`] = `
<div>
  <h1>
    HomePage 
    <small>
      John Due
    </small>
  </h1>
  <hr />
  <pre
    data-testid="code-block"
  >
    {
    "username": "John Due",
    "email": "john@email.com"
}
  </pre>
</div>
`;
